lib/Models/Cesium.ts

```diff
@@ -78,13 +79,31 @@ export default class Cesium implements GlobeOrMap {
             );
             // TODO: Look up the type in a map and call the associated function.
             //       That way the supported types of map items is extensible.
+
+            const allDataSources = allMapItems.filter(isDataSource);
+
+            // Remove deleted data sources
+            let dataSources = this.terria.dataSources;
```

Comment:
Probably a separate issue from this PR, but I think we want to move `dataSources` off `Terria` in this new architecture. It should just be owned by Cesium and Leaflet as necessary.

Reply:
`.dataSources` is now part of Cesium. Same needs to be done for Leaflet in a later PR.